README
NAME
    ITS::WICS - WICS file format converter

VERSION
    version 0.04

SYNOPSIS
        use ITS::WICS qw(xml2html);
        my $html = xml2html('path/to/file.xml');
        print $$html;

DESCRIPTION
    WICS stands for Work In Context System. As a project, it is meant to
    make Internationalization Tag Set (ITS) metadata contained in a document
    more accessible to end-users via extraction and visualization.

    This module provides access to the four WICS conversion tasks (see the
    "EXPORTS" section). Two standalone applications, a GUI and a CLI, are
    also provided in this distribution (see WICS-GUI.pl and WICS.pl).

    See "HTML 5 - ITS 2.0 IMPLEMENTATION PROJECT: WORK IN CONTEXT SYSTEM
    (WICS).pdf" in the project GitHub repository
    <https://github.com/renatb/ITS2.0-WICS-converter> for details.

EXPORTS
    The following subroutines may be exported:

  "xml2html"
    Converts input XML data into HTML5 while keeping the ITS information
    intact. See "ITS::WICS::XML2HTML" for more details.

    Argument is either a string containing an XML file name, a string
    pointer containing actual XML data, or a filehandle for a file
    containing the data.

    Return value is the newly created HTML ITS::DOM object.

  "xliff2html"
    Converts input XLIFF data into HTML5 while keeping the ITS information
    for "source" and "target" elements intact.

    The first argument is either a string containing an XLIFF file name, a
    string pointer containing actual XLIFF data, or a filehandle for a file
    containing the data. The second argument is a boolean indicating whether
    informative labels should be added (for empty or duplicate targets).

    Return value is the newly created HTML ITS::DOM object.

  "xml2xliff"
    Extracts translation units and ITS information from input XML data and
    creates an XLIFF file. This function uses "sec" elements to create
    "group"s, and "para" elements to make "trans-unit"s. Keep in mind that
    this functionality is still highly immature.

    The first argument is either a string containing an XML file name, a
    string pointer containing actual XML data, or a filehandle for a file
    containing the data.

    Return value is the newly created XLIFF ITS::DOM object.

  "reduceHtml"
    Consolidates ITS-decorated HTML5 by placing all external rules in the
    head element.

    The input and return values are the same as for "xml2HTML", except that
    the input should refer to HTML5 data instead of XML.

SEE ALSO
    This module relies on the ITS module for processing ITS markup and
    rules.

    The modules for the various ITS data conversion are included in this
    distribution:

    ITS::WICS::XML2HTML
    ITS::WICS::XLIFF2HTML
    ITS::WICS::XML2XLIFF
    ITS::WICS::Reduce
    ITS::DOM

    The ITS 2.0 specification for XML and HTML5:
    <http://www.w3.org/TR/its20/>.

    The spec for representing ITS in XLIFF:
    <http://www.w3.org/International/its/wiki/XLIFF_1.2_Mapping>.

    ITS interest group mail archives:
    <http://lists.w3.org/Archives/Public/public-i18n-its-ig/>

AUTHOR
    Nathan Glenn <garfieldnate@gmail.com>

COPYRIGHT AND LICENSE
    This software is Copyright (c) 2013 by DFKI.

    This is free software, licensed under:

      The MIT (X11) License



Changes
Revision history for ITS-WICS

{{$NEXT}}

0.04      2013-12-09 20:49:19 America/Los_Angeles
    Added support for WICS project directories.
0.04      2013-10-30 12:48:08 America/Los_Angeles
    Changed license to MIT
    Added pointer to pdf documentation in project repository
0.03      2013-10-17 13:43:53 America/Los_Angeles
    Inline elements are now properly rendered as escaped text inside <ph> elements
0.02      2013-10-16 19:06:44 America/Los_Angeles
    Removed the Wx::* prerequisites that aren't their own modules
0.01      2013-10-16 17:33:53 America/Los_Angeles
    Initial release, combining the following smaller distributions:
        ITS
        ITS-XML2HTML
        ITS-XLIFF2HTML
        ITS-XML2XLIFF
        ITS-WICS


Makefile.PL

use strict;
use warnings;

use 5.014;

use ExtUtils::MakeMaker 6.30;

use File::ShareDir::Install;
install_share dist => "share";


my %WriteMakefileArgs = (
  "ABSTRACT" => "WICS file format converter",
  "AUTHOR" => "Nathan Glenn <garfieldnate\@gmail.com>",
  "BUILD_REQUIRES" => {
    "Module::Build" => "0.3601"
  },
  "CONFIGURE_REQUIRES" => {
    "File::ShareDir::Install" => "0.03",
    "Module::Build" => "0.3601"
  },
  "DISTNAME" => "ITS-WICS",
  "EXE_FILES" => [
    "bin/WICS-GUI.pl",
    "bin/WICS.pl"
  ],
  "LICENSE" => "mit",
  "NAME" => "ITS::WICS",
  "PREREQ_PM" => {
    "Carp" => 0,
    "Data::Dumper" => 0,
    "Encode" => 0,
    "Exporter::Easy" => 0,
    "File::ShareDir::PAR" => 0,
    "Getopt::Lucid" => 0,
    "HTML::HTML5::Parser" => 0,
    "HTML::HTML5::Writer" => 0,
    "List::MoreUtils" => 0,
    "Log::Any" => 0,
    "Log::Any::Adapter" => 0,
    "Log::Any::Adapter::Base" => 0,
    "Log::Any::Adapter::Util" => 0,
    "Path::Tiny" => 0,
    "Try::Tiny" => 0,
    "URI" => 0,
    "Wx" => 0,
    "Wx::Perl::Packager" => 0,
    "XML::LibXML" => 0,
    "autodie" => 0,
    "base" => 0,
    "feature" => 0,
    "parent" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Data::Section::Simple" => 0,
    "File::Find" => 0,
    "File::Path" => 0,
    "File::Slurp" => 0,
    "File::Temp" => 0,
    "FindBin" => 0,
    "HTTP::Message" => 0,
    "IO::Compress::Gzip" => 0,
    "LWP::UserAgent" => 0,
    "Log::Any::Test" => 0,
    "Test::Base" => 0,
    "Test::Base::Filter" => 0,
    "Test::Exception" => 0,
    "Test::HTML::Differences" => 0,
    "Test::More" => "0.88",
    "Test::NoWarnings" => 0,
    "Test::Warn" => 0,
    "Test::XML" => 0,
    "utf8" => 0
  },
  "VERSION" => "0.04",
  "test" => {
    "TESTS" => "t/*.t t/ITS-Reduce/*.t t/ITS-XLIFF2HTML/*.t t/ITS-XML2HTML/*.t t/ITS-XML2XLIFF/*.t t/ITS/*.t"
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  my $tr = delete $WriteMakefileArgs{TEST_REQUIRES};
  my $br = $WriteMakefileArgs{BUILD_REQUIRES};
  for my $mod ( keys %$tr ) {
    if ( exists $br->{$mod} ) {
      $br->{$mod} = $tr->{$mod} if $tr->{$mod} > $br->{$mod};
    }
    else {
      $br->{$mod} = $tr->{$mod};
    }
  }
}

unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

{
package
MY;
use File::ShareDir::Install qw(postamble);
}


